modules/codegen/src/smithy4s/codegen/ToLine.scala

```diff
@@ -67,12 +67,26 @@ object ToLine {
         NameRef(ns, name).toLine
       case Type.Alias(_, _, aliased, _) =>
         render(aliased)
-      case Type.Ref(namespace, name)          => NameRef(namespace, name).toLine
-      case Type.PrimitiveType(prim)           => primitiveLine(prim).toLine
-      case Type.ExternalType(_, fqn, _, _, _) => NameRef(fqn).toLine
+      case Type.Ref(namespace, name) => NameRef(namespace, name).toLine
+      case Type.PrimitiveType(prim)  => primitiveLine(prim).toLine
+      case e: Type.ExternalType      => externalTypeToLine(e)
     }
   }
 
+  private[codegen] def externalTypeToLine(ext: Type.ExternalType): Line = {
+    val paramLines =
+      ext.typeParameters.map(typeToLine.render).foldLeft(Line.empty) {
+        case (acc, i) =>
+          if (acc.nonEmpty) acc + Literal(",") + Line.space + i else i
+      }
+    val nameLine = NameRef(ext.fullyQualifiedName).toLine
+    val result =
+      if (paramLines.nonEmpty)
+        nameLine + Literal("[") + paramLines + Literal("]")
+      else nameLine
+    result
+  }
+
```

Comment:
I think I can improve on this by updating NameRef to instead take in the type parameter information. Going to give it a go at least

Reply:
Done in https://github.com/disneystreaming/smithy4s/pull/649/commits/0fb6a53717fec06ccf4a1d7d8a6fb861cd5faef3